compiler/rustc_codegen_cranelift/src/base.rs

```diff
@@ -176,10 +176,10 @@ pub(crate) fn compile_fn(
         match module.define_function(codegened_func.func_id, context) {
             Ok(()) => {}
             Err(ModuleError::Compilation(CodegenError::ImplLimitExceeded)) => {
-                let handler = rustc_session::EarlyErrorHandler::new(
+                let early_dcx = rustc_session::EarlyDiagCtxt::new(
```

Comment:
Unrelated, but why are codegen backends using their own diagnostic contexts 🤔

Reply:
Good question. I've added looking into this to my errors TODO list.

Reply:
This function has a `CodegenCx`, which is constructed from a `tcx` but does not hold on to the `tcx`. So a `dcx` ref could be added to `CodegenCx` pretty easily, and then these diagnostic contexts could be removed.